packages/react-admin-core/src/stack/Switch.tsx

```diff
@@ -105,53 +109,70 @@ const StackSwitchInner: React.RefForwardingComponent<IStackSwitchApi, IProps & I
+                    url={routeProps.match.url}
+                    title={pageBreadcrumbTitle[page.props.name] || page.props.title || page.props.name}
+                >
+                    {ret}
+                </StackBreadcrumb>
+            );
+        }
+    }
 
     if (!match) return null;
 
-    return React.Children.map(props.children, (page: any) => {
-        const path = isInitialPage(page.props.name) ? `${match.url}` : `${match.url}/:id/${page.props.name}`;
-        return (
-            <Route path={path} exact={isInitialPage(page.props.name)}>
+    return (
+        <Switch>
+            {React.Children.map(props.children, (page: any) => {
+                if (isInitialPage(page.props.name)) return null; // don't render initial Page
+                const path = `${match.url}/:id/${page.props.name}`;
+                return (
+                    <Route path={path}>
+                        {(routeProps: RouteComponentProps<IRouteParams>) => {
+                            if (!routeProps.match) return null;
+                            return renderRoute(page, routeProps);
+                        }}
+                    </Route>
+                );
+            })}
+            <Route>
                 {(routeProps: RouteComponentProps<IRouteParams>) => {
-                    if (!routeProps.match) return null;
-                    activePage = page.props.name;
-                    const ret = (
-                        <StackSwitchMeta
-                            id={id}
-                            activePage={page.props.name}
-                            isInitialPageActive={isInitialPage(page.props.name)}
-                        >
-                            <StackSwitchApiContext.Provider
-                                value={api}
-                            >
-                                {typeof page.props.children === "function"
-                                    ? page.props.children(routeProps.match.params.id)
-                                    : page.props.children}
-                            </StackSwitchApiContext.Provider>
-                        </StackSwitchMeta>
-                    );
-                    if (isInitialPage(page.props.name)) {
-                        return ret;
-                    } else {
-                        return (
-                            <StackBreadcrumb
-                                url={routeProps.match.url}
-                                title={pageBreadcrumbTitle[page.props.name] || page.props.title || page.props.name}
-                            >
-                                {ret}
-                            </StackBreadcrumb>
-                        );
-                    }
+                    // now render initial page (as last route so it's a fallback)
+                    const children = React.Children.toArray(props.children) as Array<React.ReactElement<IStackPageProps>>;
+                    return renderRoute(children[0], routeProps);
```

Comment:
ok the initial page must be rendered last,
Is children[0] really always the initial page?

Reply:
you are right, that's not correct